### PR TITLE
Add basic theming to app

### PR DESCRIPTION
**Pull request by tommyjackson**
- Used the [material theme builder](https://m3.material.io/theme-builder) to generate a theme after setting just the tertiary color to be the same blue color as the new Mcs logo, so that's where this current pallet comes from. I don't think it looks half bad though at least to start.
- Created placeholder `AppTypography` and `AppShapes` objects that just use the defaults, but they're there to override if we want.
- The quickest way to get some general theming was to use material 3, and thats what had the theme builder, so that's what i went with. i haven't looked into any "platform specific look and feel" stuff, this was just to align colors and text styles and whatnot.
- brought in [Kamel](https://github.com/Kamel-Media/Kamel) for async image loading. it works, but adds a big performance hit when scrolling, so i added a convenient little boolean to the custom `TeamLogo` composable that will just use the default logos when enabled if we don't want to deal with the performance hits. but the logos make the app look so much better so i'm keeping them enabled by default
- Now all text styles, colors, and shapes should be pulling from `MaterialTheme.*`. I went through and updated the app's UI to use themed values and it is actually starting to look like a real app :') 
- fixed the iOS date time formatter since it wasn't working

Android dark mode demo: https://drive.google.com/file/d/1uzqkEwOakT6PgXlv3S0LahjXx6-CJjHz/view?usp=drive_link
iOS light mode demo: https://drive.google.com/file/d/1BqjXESddlOSbYgPDrDu0P7ipoccDMvPE/view?usp=drive_link

### CODE REVIEW

**Comment by j-roskopf** [comment]
access denied on demos jsyk

File: shared/src/commonMain/kotlin/com/boostyboys/mcs/main/MainScreen.kt [hunk 1]
```diff
@@ -30,29 +32,36 @@ class MainScreen : Screen {
...
-                        content = {
-                            Box(modifier = Modifier.padding(it)) {
-                                CurrentTab()
-                            }
-                        },
-                        bottomBar = {
-                            McsBottomNavigation()
-                        },
-                    )
+        AppTheme {
```
Comment (j-roskopf) [inline shared/src/commonMain/kotlin/com/boostyboys/mcs/main/MainScreen.kt:35]: so we set AppTheme here and in MainActivity for Android? how come?

File: shared/src/commonMain/kotlin/com/boostyboys/mcs/ui/match/MatchDetailsScreen.kt [hunk 2]
```diff
@@ -33,45 +37,67 @@ data class MatchDetailsScreen(
...
-                    ) {
-                        Spacer(modifier = Modifier.height(24.dp))
-
-                        Row(
-                            modifier = Modifier
-                                .fillMaxWidth()
-                                .padding(horizontal = 16.dp),
-                            verticalAlignment = Alignment.CenterVertically,
-                            horizontalArrangement = Arrangement.SpaceBetween,
+        AppTheme {
```
Comment (j-roskopf) [inline shared/src/commonMain/kotlin/com/boostyboys/mcs/ui/match/MatchDetailsScreen.kt:40]: do we need AppTheme here? shouldn't it be inheriting it from above?

Same comment for the other screens as well

**Review by j-roskopf — COMMENTED** [review]
did a quick lil scanachino

**Comment by tommyjackson** [comment]
> access denied on demos jsyk

should be fixed

**Pull request closed without merging**